packages/files-ui/src/Contexts/FilesContext.tsx

```diff
@@ -583,6 +621,104 @@ const FilesProvider = ({ children }: FilesContextProps) => {
         .catch(console.error)
     }, [filesApiClient, encryptForPublicKey, publicKey, refreshBuckets])
 
+  const transferFileBetweenBuckets = useCallback(async (
+    sourceBucketId: string,
+    sourceFile: FileSystemItem,
+    path: string,
+    destinationBucket: BucketKeyPermission,
+    keepOriginal = false
+  ) => {
+    const toastId = uuidv4()
+    const UPLOAD_PATH = "/"
+
+    const transferProgress: TransferProgress = {
+      id: toastId,
+      complete: false,
+      error: false,
+      progress: 0,
+      operation: "Download"
+    }
+    dispatchTransfersInProgress({ type: "add", payload: transferProgress })
+
+    getFileContent(sourceBucketId, {
+      cid: sourceFile.cid,
+      file: sourceFile,
+      path: getPathWithFile(path, sourceFile.name),
+      onDownloadProgress: (progressEvent) => {
+        dispatchTransfersInProgress({
+          type: "progress",
+          payload: {
+            id: toastId,
+            progress: Math.ceil(
+              (progressEvent.loaded / sourceFile.size) * 50
+            )
+          }
+        })
+      }
+    }).then(async (fileContent) => {
+      if (!fileContent) {
+        dispatchTransfersInProgress({
+          type: "error",
+          payload: {
+            id: toastId,
+            errorMessage: "An error occured while downloading the file"
```

Comment:
translation and typo
```suggestion
            errorMessage: t`An error occurred while downloading the file`
```